File: My JavaFullStack Project/mentorbridge_app/src/services/mockInterviewService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/mock-interviews";

// Create Interview Request
export const requestMockInterview = (data) => {
    return axios.post(`${API_URL}/request`, data);
};

// Student View Interviews
export const getStudentInterviews = (studentId) => {
    return axios.get(`${API_URL}/student/${studentId}`);
};

// Mentor View Interviews
export const getMentorInterviews = (mentorId) => {
    return axios.get(`${API_URL}/mentor/${mentorId}`);
};
// Accept Interview
export const acceptInterview = (id, meetingLink) => {
    return axios.put(
        `${API_URL}/accept/${id}?meetingLink=${encodeURIComponent(meetingLink)}`
    );
};

// Reject Interview
export const rejectInterview = (id) => {
    return axios.put(`${API_URL}/reject/${id}`);
};

File: mentorbridge_app/mentorbridge_app/src/main/java/com/example/mentorbridge_app/service/MockInterviewService.java
package com.example.mentorbridge_app.service;

import com.example.mentorbridge_app.entity.MockInterview;
import com.example.mentorbridge_app.repository.MockInterviewRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MockInterviewService {

    @Autowired
    private MockInterviewRepository repository;

    // Student creates interview request
    public MockInterview requestInterview(MockInterview interview) {

        interview.setStatus("PENDING");
        interview.setMeetingLink("");
        interview.setFeedback("");

        return repository.save(interview);
    }

    // Student views own interviews
    public List<MockInterview> getStudentInterviews(Long studentId) {

        return repository.findByStudentId(studentId);
    }

    // Mentor views interview requests
    public List<MockInterview> getMentorInterviews(Long mentorId) {

        return repository.findByMentorId(mentorId);
    }
    // Accept Interview
    public MockInterview acceptInterview(Long interviewId, String meetingLink) {

        MockInterview interview = repository.findById(interviewId)
                .orElseThrow(() -> new RuntimeException("Interview Not Found"));

        interview.setStatus("APPROVED");
        interview.setMeetingLink(meetingLink);

        return repository.save(interview);
    }

    // Reject Interview
    public MockInterview rejectInterview(Long interviewId) {

        MockInterview interview = repository.findById(interviewId)
                .orElseThrow(() -> new RuntimeException("Interview Not Found"));

        interview.setStatus("REJECTED");

        return repository.save(interview);
    }

}